frontend/app/common/fetcher.ts

```diff
@@ -13,38 +14,60 @@ export const XSRF_HEADER = 'X-XSRF-TOKEN';
 /** Cookie field with XSRF token */
 export const XSRF_COOKIE = 'XSRF-TOKEN';
 
-type Methods = typeof METHODS;
-type FetchInit = Omit<RequestInit, 'headers'> & {
-  headers?: Record<string, string>;
-  json?: unknown;
-  query?: Record<string, string | number | undefined>;
-};
-type FetcherObject = Record<Methods[number], <T>(url: string, params?: FetchInit) => Promise<T>>;
+type QueryParams = Record<string, string | number | undefined>;
+type Payload = BodyInit | Record<string, unknown> | null;
+type FetcherMethods = Record<'get' | 'delete', <T>(url: string, query?: QueryParams) => Promise<T>> &
+  Record<'put' | 'post', <T>(url: string, query?: QueryParams, body?: Payload) => Promise<T>>;
+
+export function stringifyUrl(uri: string, query: QueryParams) {
+  const queryEntries = Object.entries(query);
+  const siteIdParam = `site=${encodeURIComponent(siteId)}`;
+  const baseUrl = uri.indexOf('/auth') === 0 ? BASE_URL : `${BASE_URL}${API_BASE}`;
+  const url = `${baseUrl}${uri}`;
+
+  if (queryEntries.length === 0) {
+    return `${url}?${siteIdParam}`;
+  }
+
+  const queryString = Object.entries(query)
+    .reduce(
+      (accum, [k, v]) => (v === undefined ? accum : [...accum, `${encodeURIComponent(k)}=${encodeURIComponent(v)}`]),
+      [siteIdParam] as string[]
+    )
+    .join('&');
+
+  return `${url}?${queryString}`;
+}
 
 /** JWT token received from server and will be send by each request, if it present */
 let activeJwtToken: string | undefined;
 
-const fetcher = METHODS.reduce((acc, method) => {
-  acc[method] = async (uri: string, params: FetchInit = {}) => {
-    const { headers = {}, json, ...fetchParams } = params;
+const fetcher = METHODS.reduce<FetcherMethods>((acc, method) => {
+  acc[method] = async (uri: string, query: QueryParams = {}, body?: Payload) => {
     // add api base if it's not auth request
     // we use `indexOf` instead of `startsWidth` because we don't want to have another one polyfill for no reason
-    const baseUrl = uri.indexOf('/auth') === 0 ? BASE_URL : `${BASE_URL}${API_BASE}`;
-    const url = `${baseUrl}${uri}`;
+    const url = stringifyUrl(uri, query);
```

Comment:
Inconsistent use of BASE_URL across the changed code looks weird, as well as appending it here.

Reply:
It was the same but determined by passed param.
We know we don't need to add API_BASE for auth requests and determine it by start of URI

Reply:
Added a comment in the more relevant place of the code.